pkg/events/queue/queue_mem_list.go

```diff
@@ -0,0 +1,114 @@
+package queue
+
+import (
+	"container/list"
+	"fmt"
+	"github.com/aquasecurity/tracee/pkg/external"
+	"sync"
+)
+
+type EventQueueMem struct {
+	cache                *list.List
+	mutex                sync.Mutex
+	maxAmountOfEvents    int // number of cached events
+	EventsCacheMemSizeMB int
+}
+
+func (q *EventQueueMem) String() string {
+	return fmt.Sprintf("In-Memory Event Queue (Size = %d MB)", q.EventsCacheMemSizeMB)
+}
+
+func (q *EventQueueMem) Setup() error {
+	q.maxAmountOfEvents = q.getQueueSizeInEvents()
+	q.cache = new(list.List)
+
+	return nil
+}
+
+// Enqueue pushes an event into the queue (may block until queue is available)
+func (q *EventQueueMem) Enqueue(evt *external.Event) {
+	if evt == nil {
+		return
+	}
+
+	// "spin" until cache length is smaller then maxAmountOfEvents (blocking)
```

Comment:
spinning will cause high cpu usage. I am not sure why we ain't using condition variable. wdyt?

Reply:
Like I said, splitting the condition variable from the transaction in charge for changing the actual data is racy (my previous comment). I suggested a change to reduce CPU consumption, let me know what you think.

Reply:
Per last comment in the other comment.